### PR TITLE
[mini] move typecheck related code to separate file

### PR DESCRIPTION
**Pull request by lewurm**
for public functions I added the `typechecks_` prefix.  any objections?

### CODE REVIEW

**Comment by vargaz** [comment]
We usually use the mini_ prefix for jit functions.

**Comment by lewurm** [comment]
sorry for the late follow-up.  updated exported functions with `mini_` prefix.

File: mono/mini/Makefile.am.in [hunk 1]
```diff
@@ -440,6 +440,8 @@ common_sources = \
 	mini-exceptions.c	\
 	mini-exceptions-native-unwinder.c	\
 	mini-trampolines.c  	\
+	typechecks.h		\
+	typechecks.c		\
```
Comment (akoeplinger) [inline mono/mini/Makefile.am.in]: Please add these to the appropriate MSVC .vcxproj

**Comment by lewurm** [comment]
superseded by #4048

**Pull request closed without merging**